### PR TITLE
Add extended information about XCom backends in docs

### PR DESCRIPTION
**Pull request by turbaszek**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: docs/apache-airflow/concepts.rst [hunk 1]
```diff
@@ -770,35 +770,131 @@ passed, then a corresponding list of XCom values is returned.
...
+.. code-block:: python
+
+    "SELECT * FROM {{ task_instance.xcom_pull(task_ids='foo_task', key='table_name') }}"
+
+This can also be done using the ``output`` attribute of an operator. The above example can be then
+simplified:
+
+.. code-block:: python
+
+    f"SELECT * FROM { foo_task.output['table_name'] }"
```
Comment (casassg) [inline docs/apache-airflow/concepts.rst:785]: Should we document this ``output`` better somewhere?

File: docs/apache-airflow/concepts.rst [hunk 2]
```diff
@@ -770,35 +770,131 @@ passed, then a corresponding list of XCom values is returned.
...
+To use a custom XCom backend users should configure ``xcom_backend`` parameter in Airflow config. Provided value
+should point to a class that is subclass of :class:`~airflow.models.xcom.BaseXCom` (See :doc:`modules_management` for
+details on how Python and Airflow manage modules).
 
-It is possible to change ``XCom`` behaviour of serialization and deserialization of tasks' result.
-To do this one have to change ``xcom_backend`` parameter in Airflow config. Provided value should point
-to a class that is subclass of :class:`~airflow.models.xcom.BaseXCom`. To alter the serialization /
-deserialization mechanism the custom class should override ``serialize_value`` and ``deserialize_value``
-methods.
+A custom XCom class has to implement the tow following methods:
```
Comment (casassg) [inline docs/apache-airflow/concepts.rst:820]:
```suggestion
A custom XCom class has to implement the two following methods:
```

**Review by casassg — APPROVED** [review]
lgtm!

**Comment by mik-laj** [comment]
Have you thought about moving some documentation to a new document? I think this description goes beyond the general concepts of Airflow.

File: docs/apache-airflow/concepts.rst (hunk 1 above)
Comment (turbaszek) [reply docs/apache-airflow/concepts.rst:785]: Yes, we should. I was to tired to write more about XComArgs

**Comment by turbaszek** [comment]
> Have you thought about moving some documentation to a new document? I think this description goes beyond the general concepts of Airflow.

Any suggestions?

**Comment by mik-laj** [comment]
Example: https://github.com/apache/airflow/pull/10241/files

**Pull request closed without merging**